network/tcp_test.go

```diff
@@ -0,0 +1,540 @@
+package network
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/crypto/config"
+	"github.com/stretchr/testify/require"
+)
+
+type BigMsg struct {
+	Array []byte
+}
+
+type fakeConn struct {
+	// how many bytes does it write at maximum at each call
+	max int
+	// do we fail on the first write
+	fail1 bool
+	done1 bool
+	// do we fail on every successive write
+	failRest bool
+	// how many total bytes have we written
+	writtenBytes int
+}
+
+type fakeAddr string
+
+func (f fakeAddr) Network() string {
+	return "network"
+}
+
+func (f fakeAddr) String() string {
+	return "network-string"
+}
+
+func (f *fakeConn) Read(b []byte) (n int, e error) {
+	return 0, nil
+}
+
+func (f *fakeConn) Write(b []byte) (n int, e error) {
+	if !f.done1 && f.fail1 {
+		return 0, ErrUnknown
+	} else if f.failRest {
+		return 0, ErrUnknown
+	}
+	if len(b) < f.max {
+		f.writtenBytes += len(b)
+		return len(b), nil
+	}
+	f.writtenBytes += f.max
+	return f.max, nil
+}
+
+func (f *fakeConn) Close() error {
+	return nil
+}
+
+func (f *fakeConn) LocalAddr() net.Addr {
+	return fakeAddr("")
+}
+func (f *fakeConn) RemoteAddr() net.Addr {
+	return fakeAddr("")
+}
+
+func (f *fakeConn) SetDeadline(t time.Time) error {
+	return nil
+}
+
+func (f *fakeConn) SetReadDeadline(t time.Time) error {
+	return nil
+}
+func (f *fakeConn) SetWriteDeadline(t time.Time) error {
```

Comment:
I wonder whether it would work to embed `TCPConn` into `fakeConn` and then don't re-define those messages, but only the ones you want to change? Would make it more readable.